code/modules/mob/living/living.dm

```diff
@@ -1127,18 +1127,31 @@ default behaviour is:
 		if ((!( yes ) || now_pushing) || !loc)
 			return
 		now_pushing = 1
-		if (istype(AM, /mob/living))
+		if (istype(AM, /obj/structure/bed/roller)) //no pushing rollerbeds that have people on them
+			var/obj/structure/bed/roller/R = AM
+			for(var/mob/living/tmob in range(R, 1))
+				if(tmob.pulling == R && !(tmob.restrained()) && tmob.stat == 0 && R.density == 1)
+					if (!(world.time % 5))
```

Comment:
That would require entirely different code, this is designed to make it shoot off with a 20% probability